src/main/java/de/tum/in/www1/artemis/service/connectors/athena/AthenaFeedbackSuggestionsService.java

```diff
@@ -90,4 +98,26 @@ public List<ProgrammingFeedbackDTO> getProgrammingFeedbackSuggestions(Programmin
         log.info("Athena responded to feedback suggestions request: {}", response.data);
         return response.data.stream().toList();
     }
+
+    /**
+     * Calls the remote Athena service to get feedback suggestions for a given submission.
+     *
+     * @param exercise   the {@link ModelingExercise} the suggestions are fetched for
+     * @param submission the {@link ModelingSubmission} the suggestions are fetched for
+     * @return a list of feedback suggestions
+     */
+    public List<ModelingFeedbackDTO> getModelingFeedbackSuggestions(ModelingExercise exercise, ModelingSubmission submission) throws NetworkingException {
+        log.debug("Start Athena Feedback Suggestions Service for Exercise '{}' (#{}).", exercise.getTitle(), exercise.getId());
+
+        if (!Objects.equals(submission.getParticipation().getExercise().getId(), exercise.getId())) {
+            log.error("Exercise id {} does not match submission's exercise id {}", exercise.getId(), submission.getParticipation().getExercise().getId());
+            throw new ConflictException("Exercise id " + exercise.getId() + " does not match submission's exercise id " + submission.getParticipation().getExercise().getId(),
+                    "Exercise", "exerciseIdDoesNotMatch");
+        }
+
+        final RequestDTO request = new RequestDTO(athenaDTOConverterService.ofExercise(exercise), athenaDTOConverterService.ofSubmission(exercise.getId(), submission));
```

Comment:
if I understand correctly, `athenaDTOConverterService.ofExercise()` will reload the modelling exercise again from the database, alongside grading criteria, while the modelling exercise is already loaded? can't this process be optimised with only querying grading criteria of this modelling exercise instead of doing a query with a join?